R/html_resources.R

```diff
@@ -269,8 +277,7 @@ discover_rmd_resources <- function(rmd_file, encoding,
               recursive = TRUE,
               include.dirs = FALSE)
             lapply(files, function(f) {
-              list(path = file.path(explicit_res$path, f), 
-                   explicit = TRUE,
+              discover_single_resource(file.path(explicit_res$path, f), TRUE,
```

Comment:
You're really pretty close to having the globbing and directory cases the same. What if you consider all of this as having two halves:

1. identify the file or files referenced by the path or paths.
2. call `discover_single_resource` on each.

This would probably leave us with a single `discover_single_resource` call and maybe easier-to-understand code?